system/mp/mpcomm.cpp

```diff
@@ -2817,6 +2820,7 @@ class CCommunicator: public ICommunicator, public CInterface
     CMPServer *parent;
     bool outer;
     rank_t myrank;
+    CriticalSection verifysect;
```

Comment:
NB: moved from global scope to instance scope.
It protects verifyAll/verifyConnection/disconnect from one another, I can't see why it was at global scope, and my test involved multiple overlapping verifyAll's in fact, I think it was only luck that worked before because of the way that verifyAll backs off for 100ms in a loop.

But please look/consider any possible issues here.